backend/src/services/llm_parser.py

```
@@ -275,6 +283,64 @@ async def _parse_with_anthropic(self, prompt: str) -> Dict:
+                        json_start = result_text.find('{')
+                        json_end = result_text.rfind('}') + 1
+                        if json_start >= 0 and json_end > json_start:
+                            json_str = result_text[json_start:json_end]
+                            parsed_data = json.loads(json_str)
+                            parsed_data["source"] = "perplexity"
+                            return parsed_data
+                        else:
+                            try:
+                                parsed_data = json.loads(result_text)
+                                parsed_data["source"] = "perplexity"
+                                return parsed_data
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)